actor.go

```diff
@@ -69,3 +69,14 @@ func (actor *Actor) IsLeader() bool {
 	}
 	return false
 }
+
+// Leader returns the LibP2P ID of the Raft leader.
+func (actor *Actor) Leader() peer.ID {
+	// Leader as returned by Libp2pTransport.LocalAddr()
+	raftLeader := actor.Raft.Leader()
+	peerID, err := peer.IDB58Decode(raftLeader)
+	if err != nil {
+		panic(err)
```

Comment:
I had that same feeling. That string comes from the `LocalAddr()` in transport.go in the end, which does a `peer.IDB58Encode(t.host.ID())`. If this is invalid here something is really badly broken somewhere else. I'm happy to blow up the whole thing in that case.